### PR TITLE
Simplify Docker SSL handling

### PR DESCRIPTION
**Pull request by williamjallen**
CDash currently requires SSL/TLS by default when using our production image.  While useful for Docker Compose-based systems, this is a major pain point for k8s systems which handle TLS termination at ingress.  See https://github.com/Kitware/CDash/issues/2179, for example.  This PR aims to ease these pains by conditionally enabling SSL/TLS if certificates are provided, in preparation for upcoming work to better support k8s-based deployments.

Fixes https://github.com/Kitware/CDash/issues/2179.

### CODE REVIEW

**Comment by williamjallen** [comment]
@zackgalbreath While the goal was to assist my Helm chart work, this is still very much a draft and I suspect that this may overlap with your work with the UBI image deployment.  Feel free to push changes to this PR if you have any suggestions.

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ services:
     environment:
       DB_HOST: database
     healthcheck:
-      test: curl -s -o /dev/null -w "%{http_code}" http://cdash:8080/ping | grep 200 > /dev/null
+      test: curl -s -o /dev/null -w "%{http_code}" http://cdash:80/ping | grep 200 > /dev/null
```
Comment (zackgalbreath) [inline docker/docker-compose.yml:15]: This healthcheck doesn't work for a production docker compose installation now.

Experimenting locally, I had to do:
`curl -k https://cdash:80/ping`

...but the `-k` may be specific to this particular instance.

**Comment by williamjallen** [comment]
The tests require the exposed port in the dev environment to match inside & outside the container, in addition to the issues observed by @zackgalbreath with the UBI image.  As a result, I switched the port back to 8080.

@zackgalbreath does this resolve the health check issue, or are the certificates the issue?

**Comment by zackgalbreath** [comment]
> @zackgalbreath does this resolve the health check issue, or are the certificates the issue?

The containers spin up successfully but the healthcheck issue remains. Reproducing this manually, the relevant error message is:

```
Bad Request

Your browser sent a request that this server could not understand.
Reason: You're speaking plain HTTP to an SSL-enabled server port.
Instead use the HTTPS scheme to access this URL, please.
```

**Review by zackgalbreath — APPROVED** [review]

**Pull request merged**